### PR TITLE
docs update

### LINKED ISSUE #21: Documentation needs update to reflect new class structure

### PR DESCRIPTION
**Pull request by msarahan**
fixes #21

### CODE REVIEW

**Comment by msarahan** [comment]
@youngblood please take a look.  When you're happy with content let me know and I will fix merge conflicts and merge.

**Comment by msarahan** [comment]
PS: view docs at http://topik.readthedocs.org/en/msarahan-docs_update/index.html

File: docs/example.rst [hunk 1]
```diff
@@ -1,46 +1,102 @@
...
 
+An example complete workflow would be the following:
 
+.. code-block:: python
 
-Analyzing the results
----------------------
+   >>> from topik import read_input, registered_models
+   >>> raw_data = read_input("reviews")
+   >>> raw_data.tokenize()
```
Comment (youngblood) [inline docs/example.rst]: need to assign to new variable

File: docs/example.rst [hunk 2]
```diff
@@ -1,46 +1,102 @@
...
+   >>> n_topics = 10
+   >>> model = registered_models["LDA"](raw_data, n_topics)
```
Comment (youngblood) [inline docs/example.rst]: use new variable from above

File: docs/read_input.rst [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                           json_prefix="nested.dictionary")
+
+
+`Elasticsearch` additional options and notes
+============================================
+
+The `Elasticsearch` importer expects a full string specifying the `Elasticsearch`
+server. This string at a minimum must contain both the server address and the
+index to access (if any). All results returned from the `Elasticsearch` query
+contain only the contents of the '_source' field returned from the query.
```
Comment (youngblood) [inline docs/read_input.rst]: index has to be set separately

File: docs/read_input.rst [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+creation. This can be used to pass additional login parameters, for example, to
+use SSL:
+
+.. code-block:: python
+
+   >>> corpus = read_input(source="https://user:secret@localhost:443/test_index",
+                           content_field="text", use_ssl=True)
+
+
+The source argument for Elasticsearch also supports multiple servers, though
```
Comment (youngblood) [inline docs/read_input.rst]: need to actually specify source_type in example

File: docs/read_input.rst [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                            content_field="text")
+
+
+For more information on server options, please refer to `Elasticsearch's
+documentation <https://elasticsearch-py.readthedocs.org/en/master/>`_.
+
+Extra keyword arguments are also passed to the scroll helper that returns
+results. Of special note here, an additional ``query`` keyword argument can be
+passed to limit the records imported from the server. This query must follow the
+Elasticsearch query of filter DSL. For more information on Elasticsearch query
```
Comment (youngblood) [inline docs/read_input.rst]: remove "of filter"

File: docs/read_input.rst [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+The default output is the :class:`~.DictionaryCorpus`. No additional arguments
+are necessary. :class:`~.DictionaryCorpus` stores everything in a Python
+dictionary. As such, it is memory intensive. All operations done with a
+:class:`~.DictionaryCorpus` block until complete. :class:`~.DictionaryCorpus` is
+the simplest to use, but it will ultimately limit the size of analyses that you
+can perform.
+
+The :class:`~.ElasticSearchCorpus` can be specified
+to :func:`~.read_input` using the ``output_type`` argument. It must
```
Comment (youngblood) [inline docs/read_input.rst]: this needs to match "output_format" below

File: docs/read_input.rst [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    >>> output_args = {"source": "localhost", "index": "destination_index"}
+    >>> raw_data = read_input("test_data.json", output_format='elastic',
+                              output_args=output_args, content_field="text")
+
+
+:class:`~.ElasticSearchCorpus` stores everything in an `Elasticsearch` instance
+that you specify. Operations do not block, and have "eventual consistency": the
+corpus will eventually have all of the documents you sent available, but not
+necessarily immediately after the read_input function returns. This lag time is
+due to `Elasticsearch` indexing the data on the server side.
```
Comment (youngblood) [inline docs/read_input.rst]: add section to describe the synchronous wait function

File: docs/tokenization.rst [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    Gensim.
+  * "collocation": Collects bigrams and trigrams in addition to single words.
+    Uses NLTK.
+  * "entities": Extracts noun phrases as entities. Uses NLTK.
+  * "mixed": first extracts noun phrases as entities, then follows up with
+    simple tokenization for single words. Uses NLTK.
+
+All methods accept a keyword argument ``stopwords``, which are words that will
+be ignored in tokenization. These are words that add little content value, such
+as prepositions. The default, STOPWORDS, uses gensim's STOPWORDS collection.
```
Comment (youngblood) [inline docs/tokenization.rst]: Default is actually None, which uses gensim's default stopwords

File: docs/tokenization.rst [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+The tokenize method accepts a few arguments to specify a tokenization method and
+control behavior therein. The available tokenization methods are available in
+the :data:`~.tokenizer_methods` dictionary. The presently available
+methods are:
+
+  * "simple": (default) lowercases input text and extracts single words. Uses
+    Gensim.
+  * "collocation": Collects bigrams and trigrams in addition to single words.
+    Uses NLTK.
+  * "entities": Extracts noun phrases as entities. Uses NLTK.
```
Comment (youngblood) [inline docs/tokenization.rst]: uses TextBlob, not NLTK

File: docs/tokenization.rst [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+  * "mixed": first extracts noun phrases as entities, then follows up with
```
Comment (youngblood) [inline docs/tokenization.rst]: uses TextBlob, not NLTK

File: docs/tokenization.rst [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+.. code-block:: python
+
+   >>> from topik.tokenizers import collect_bigrams_and_trigrams
+   >>> patterns = collect_bigrams_and_trigrams(corpus)
+
+
+Parameterization is done at this step, prior to tokenization of the corpus.  Tweakable parameters are:
+
+  * top_n: limit results to a maximum number
+  * min_word_length: the minimum length that any single word can be
```
Comment (youngblood) [inline docs/tokenization.rst]: should be 'min_length'

File: docs/tokenization.rst [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+##########################
+
+The next step in topic modeling is to break your documents up into individual
+terms. This is called tokenization. Tokenization is done using the
+:meth:`~.CorpusInterface.tokenize` method on a Corpus object
+(returned from :func:`~.read_input`):
+
+.. code-block:: python
+
+   >>> raw_data.tokenize()
```
Comment (youngblood) [inline docs/tokenization.rst]: assign to new viariable

File: docs/tokenization.rst [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+For small bodies of text, you'll need small freq values, but this may be
+correspondingly "noisy."
+
+Next, feed the patterns into the :meth:`~.CorpusInterface.tokenize` method of
+your corpus object:
+
+.. code-block:: python
+
+   >>> raw_data.tokenize(method="collocation", patterns=patterns)
```
Comment (youngblood) [inline docs/tokenization.rst]: assign to new viariable

File: docs/tokenization.rst [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+   >>> entities = collect_entities(corpus, freq_min=4, freq_max=10000)
+
+
+Next, tokenize the document collection:
+
+
+.. code-block:: python
+
+   >>> raw_data.tokenize(method="entities", entities=entities)
```
Comment (youngblood) [inline docs/tokenization.rst]: assign to new viariable

File: docs/tokenization.rst [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+
+Mixed tokenization employs both the entities tokenizer and the simple tokenizer,
+for when the entities tokenizer is overly restrictive, or for when words are
+interesting both together and apart. Usage is similar to the entities tokenizer:
+
+.. code-block:: python
+
+   >>> from topik.tokenizers import collect_entities
+   >>> entities = collect_entities(corpus)
+   >>> raw_data.tokenize(method="mixed", entities=entities)
```
Comment (youngblood) [inline docs/tokenization.rst]: assign to new viariable

File: docs/topic_modeling.rst [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+(sparse) matrix of which documents contain which words, attempting to identify
+latent "topics". At the end of modeling, each document will have a mix of topics
+that it belongs to, each with a weight. Each topic in turn will have weights
+associated with the collection of words from all documents.
+
+Currently, Topik provides interfaces to or implements two topic modeling
+algorithms, LDA (latent dirichlet allocation) and PLSA (probablistic latent
+semantic analysis). LDA and PLSA are closely related, with LDA being a slightly
+more recent development. The authoritative sources on LDA and PLSA are `Blei,
+Ng, Jordan (20003) <http://jmlr.csail.mit.edu/papers/v3/blei03a.html>`_, and
```
Comment (youngblood) [inline docs/topic_modeling.rst]: 2003

File: topik/intermediaries/raw_data.py [hunk 17]
```diff
@@ -177,6 +176,7 @@ def import_from_iterable(self, iterable, id_field="text", batch_size=500):
             list of strings, a dictionary with one key, "text" is created and
             used.
         """
+        from elasticsearch import helpers
         batch = []
         for item in iterable:
             if isinstance(item, basestring):
```
Comment (youngblood) [inline topik/intermediaries/raw_data.py]: document Corpus saving and loading (this particular line not relevant to comment)

File: docs/example.rst [hunk 18]
```diff
@@ -1,46 +1,102 @@
...
+options available to you for each step.
 
+An example complete workflow would be the following:
 
+.. code-block:: python
 
-Analyzing the results
----------------------
+   >>> from topik import read_input, registered_models
+   >>> raw_data = read_input("reviews")
```
Comment (youngblood) [inline docs/example.rst]: This gives an error that an additional argument is required.

Comment (youngblood) [inline docs/example.rst]: should read_input's content_field parameter have a default of None as does run_model?

Comment (msarahan) [inline docs/example.rst]: No, my humble opinion is that this is truly required.  The probability of failure when this information is not provided or not accurate is practically 100%.  If anything, run_model should add the requirement.

**Comment by msarahan** [comment]
@youngblood please review these changes, and merge when ready.

**Pull request merged**